### PR TITLE
Efrat/lbm1 6987 lightfield info

### PR DESCRIPTION
**Pull request by efrat-lb**
a tool to determine if there is lightfield basicly and on which numa,
extra data regarding the update date, and HW dept knowledge.
selftest Get this information - under lightfield
    "lightfield": {
      "lspci": {
        "04:00": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "05:00": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "05:01": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "05:04": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "05:08": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "05:09": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "06:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "07:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "16:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "17:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "81:00": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "82:00": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "82:01": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "82:04": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "82:08": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "82:09": "PCI bridge: PLX Technology, Inc. Device 8764 (rev ab)",
        "83:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "84:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "93:00": "Serial controller: Device 1d9a:0001 (rev 01)",
        "94:00": "Serial controller: Device 1d9a:0001 (rev 01)"
      },
      "numa0": {
        "00": "02",
        "01": "01",
        "02": "10",
        "03": "29",
        "04": "18",
        "05": "10",
        "06": "29",
        "07": "02",
        "08": "ae",
        "09": "0a",
        "10": "01",
        "13": "ff",
        "14": "ff",
        "15": "ff",
        "16": "01",
        "17": "18",
        "18": "10",
        "19": "22",
        "s/n": "0318"
      },
      "numa1": {
        "00": "02",
        "01": "18",
        "02": "10",
        "03": "29",
        "04": "18",
        "05": "10",
        "06": "29",
        "07": "02",
        "08": "ae",
        "09": "0a",
        "10": "01",
        "13": "ff",
        "14": "ff",
        "15": "ff",
        "16": "01",
        "17": "18",
        "18": "10",
        "19": "22",
        "s/n": "0319"

### CODE REVIEW

**Comment by efrat-lb** [comment]
you can check under:
inaugurator.thin.initrd.img.1
inaugurator.vmlinuz.1
thanks

File: Makefile.build [hunk 1]
```diff
@@ -120,6 +120,7 @@ build/inaugurator.thin.initrd.dir: ${PYTHON_DIST_LOCAL_PATH} build/osmosis-1.0.l
 	DEST=$@.tmp sh/relative_copy_executable.sh /usr/sbin/lshw
 	DEST=$@.tmp sh/relative_copy_executable.sh /usr/sbin/lspci
 	DEST=$@.tmp sh/relative_copy_executable.sh /usr/bin/lscpu
+	DEST=$@.tmp sh/relative_copy_executable.sh /root/inaugurator/inaugurator/execs/VPD
```
Comment (yuval-lb) [inline Makefile.build]: I think the VPD tool is missing in the PR

Comment (yuval-lb) [reply Makefile.build]: ah, I missed the BIN file.

File: inaugurator/hwinfo.py [hunk 2]
```diff
@@ -64,6 +64,40 @@ def get_memory():
...
+        return {'error': e.message}
+
+def get_lightfield(numa):
+    r = ""
+    try:
+        r = sh.run("/root/inaugurator/inaugurator/execs/VPD -r 20 -n %s" % str(numa)).strip()
+        if not r or 'found no device' in r:
+            return 'VPD failed'
+        elements = r.split(':')[-1]
+        d = {}
```
Comment (yuval-lb) [inline inaugurator/hwinfo.py]: what is d?

File: inaugurator/hwinfo.py [hunk 3]
```diff
@@ -64,6 +64,40 @@ def get_memory():
...
+            port, val = line.split(".", 1)
+            d[str(port).strip()] = val[2:]
+        return d
+    except Exception as e:
+        return {'error': e.message}
+
+def get_lightfield(numa):
+    r = ""
+    try:
+        r = sh.run("/root/inaugurator/inaugurator/execs/VPD -r 20 -n %s" % str(numa)).strip()
```
Comment (yuval-lb) [inline inaugurator/hwinfo.py]: r?

Comment (yuval-lb) [inline inaugurator/hwinfo.py]: can we send the output to selftest and do the logic over there for maintenance reasons?

File: Makefile.build (hunk 1 above)
Comment (sashas-lb) [reply Makefile.build]: hosestly guys  i think it is a bad bad idea to feed inaugurator with all those tools ... what is we will have 2 versions of lightfield ? 2 inaugurators ? several versions with tool ? I think you are shooting yourself with a foot with this design.
It starts to look like more than just "small provisioning" tool

File: inaugurator/hwinfo.py [hunk 4]
```diff
@@ -64,6 +64,40 @@ def get_memory():
         return {'error': e.message}
 
 
+def get_lspci_lf():
+    '''
+    lspci indicate if exist and if lightfield is overpassed
+    '''
+    try:
+        r = sh.run('lspci|grep -iE "8764|1d9a"')
```
Comment (sashas-lb) [inline inaugurator/hwinfo.py]: please for god sake document what the hell are those PCI ids.
@ofer-lb  for next gen LF will it be the same ?

Comment (ofer-lb) [reply inaugurator/hwinfo.py]: yes

Comment (efrat-lb) [reply inaugurator/hwinfo.py]: the lspci will find lightfield from older Versions as oppose to VPD.
you want the VPD out ?
I suggest the next phase will be to understand if a X machine deserve the special label or not, we can decide even to put X in an error state.
speaking with Yossi I realize we can detect errors.

Comment (yuval-lb) [reply inaugurator/hwinfo.py]: I suggest removing the VPD for now. if  our target only need is to identify LF in machines, and we can know it from lspci, then we dont want to add extra noise.

File: inaugurator/hwinfo.py (hunk 3 above)
Comment (efrat-lb) [reply inaugurator/hwinfo.py]: yes, done

File: inaugurator/hwinfo.py (hunk 4 above)
Comment (efrat-lb) [reply inaugurator/hwinfo.py]: updated output:
>"lightfield": {
>   "lspci": { .. },
>   "numa0": {"Read parameters": "02 18 10 29 18 10 29 02 ae 0a 01 15 03 ff ff ff 01 18 10 22"},
>   "numa1": {  "Read parameters": "02 18 10 29 18 10 29 02 ae 0a 01 29 03 ff ff ff 01 18 10 22" }
>     }

File: inaugurator/hwinfo.py [hunk 5]
```diff
@@ -92,17 +92,10 @@ def get_lightfield(numa):
...
-        d['s/n'] = stream[12] + stream[11]  # lsb 11
-        for i, v in enumerate(stream):
-            if i in [11,12]:
-                continue
-            d["%02d" % i]= v.strip()
-        return d
+        header, registers = r.split(':',1)
+        return {header.strip() : registers.strip()}
     except Exception as e:
         return {'error': e.message}
```
Comment (yuval-lb) [inline inaugurator/hwinfo.py]: squash latest 3 commits, verify it works, and its good to go

**Pull request merged**